commit 22312e08e7923a39668d58060b063bb9e325edb6 (refs/changes/39/3739/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2018-03-29T14:46:19-07:00 (1 year, 6 months ago)
    
    TOOLS-1950 sdcnode 4.x and higher should restore the ability to reload TLS certs at runtime
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/nodeconfigs.json b/nodeconfigs.json
index 77bf979..d7cdb18 100644
--- a/nodeconfigs.json
+++ b/nodeconfigs.json
@@ -3,6 +3,9 @@
         "// image": "sdc-minimal-multiarch-lts 15.4.1",
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "version": "v8.9.3",
+        "patch_files": [
+            "patches/tls-reload-certs-v8.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "gcc_version_prefix": "4.9.",
         "rpath": "$ORIGIN/../lib",
@@ -14,6 +17,9 @@
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "comment": "For use by services running in zones. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
         "version": "v8.9.3",
+        "patch_files": [
+            "patches/tls-reload-certs-v8.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "build_tag": "zone"
     },
@@ -22,6 +28,9 @@
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "comment": "For use by services running in zones. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
         "version": "v8.9.3",
+        "patch_files": [
+            "patches/tls-reload-certs-v8.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=x64 --with-dtrace",
         "build_tag": "zone64"
     },
@@ -30,6 +39,9 @@
         "// image": "sdc-minimal-multiarch-lts 15.4.1",
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "version": "v6.12.2",
+        "patch_files": [
+            "patches/tls-reload-certs-v6.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "gcc_version_prefix": "4.9.",
         "rpath": "$ORIGIN/../lib",
@@ -41,6 +53,9 @@
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "comment": "For use by services running in zones. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
         "version": "v6.12.2",
+        "patch_files": [
+            "patches/tls-reload-certs-v6.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "build_tag": "zone"
     },
@@ -49,6 +64,9 @@
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "comment": "For use by services running in zones. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
         "version": "v6.12.2",
+        "patch_files": [
+            "patches/tls-reload-certs-v6.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=x64 --with-dtrace",
         "build_tag": "zone64"
     },
@@ -57,6 +75,9 @@
         "// image": "sdc-minimal-multiarch-lts 15.4.1",
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "version": "v4.8.7",
+        "patch_files": [
+            "patches/tls-reload-certs-v4.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "gcc_version_prefix": "4.9.",
         "rpath": "$ORIGIN/../lib",
@@ -68,6 +89,9 @@
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "comment": "For use by services running in zones. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
         "version": "v4.8.7",
+        "patch_files": [
+            "patches/tls-reload-certs-v4.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "build_tag": "zone"
     },
@@ -76,6 +100,9 @@
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "comment": "For use by services running in zones. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
         "version": "v4.8.7",
+        "patch_files": [
+            "patches/tls-reload-certs-v4.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=x64 --with-dtrace",
         "build_tag": "zone64"
     },
@@ -84,6 +111,9 @@
         "// image": "sdc-minimal-multiarch-lts 15.4.1",
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "version": "v4.8.5",
+        "patch_files": [
+            "patches/tls-reload-certs-v4.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "gcc_version_prefix": "4.9.",
         "rpath": "$ORIGIN/../lib",
@@ -95,6 +125,9 @@
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "comment": "For use by services running in zones. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
         "version": "v4.8.5",
+        "patch_files": [
+            "patches/tls-reload-certs-v4.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "build_tag": "zone"
     },
@@ -103,6 +136,9 @@
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "comment": "For use by services running in zones. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
         "version": "v4.8.5",
+        "patch_files": [
+            "patches/tls-reload-certs-v4.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=x64 --with-dtrace",
         "build_tag": "zone64"
     },
@@ -111,6 +147,9 @@
         "// image": "sdc-minimal-multiarch-lts 15.4.1",
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "version": "v4.6.1",
+        "patch_files": [
+            "patches/tls-reload-certs-v4.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "gcc_version_prefix": "4.9.",
         "rpath": "$ORIGIN/../lib",
@@ -121,6 +160,9 @@
         "// image": "sdc-minimal-multiarch-lts 15.4.1",
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "comment": "For use by services running in zones. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
+        "patch_files": [
+            "patches/tls-reload-certs-v4.x.patch"
+        ],
         "version": "v4.6.1",
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "build_tag": "zone"
@@ -130,6 +172,9 @@
         "// image": "sdc-minimal-multiarch-lts 15.4.1",
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "version": "v4.6.0",
+        "patch_files": [
+            "patches/tls-reload-certs-v4.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "gcc_version_prefix": "4.9.",
         "rpath": "$ORIGIN/../lib",
@@ -141,6 +186,9 @@
         "image": "18b094b0-eb01-11e5-80c1-175dac7ddf02",
         "comment": "For use by services running in zones. Statically link to OpenSSL and zlib *for now* until OS-1294/TOOLS-130.",
         "version": "v4.6.0",
+        "patch_files": [
+            "patches/tls-reload-certs-v4.x.patch"
+        ],
         "configure_flags": "--without-snapshot --dest-cpu=ia32 --with-dtrace",
         "build_tag": "zone"
     },
diff --git a/patches/tls-reload-certs-v4.x.patch b/patches/tls-reload-certs-v4.x.patch
new file mode 100644
index 0000000..91e473d
--- /dev/null
+++ b/patches/tls-reload-certs-v4.x.patch
@@ -0,0 +1,14 @@
+TOOLS-1950 sdcnode 4.x should restore the ability to reload TLS certs at runtime
+diff --git a/src/node_crypto.cc b/src/node_crypto.cc
+index 13af291dfc..b90f014bd6 100644
+--- a/src/node_crypto.cc
++++ b/src/node_crypto.cc
+@@ -494,7 +494,7 @@ int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
+     // the CA certificates.
+     int r;
+
+-    SSL_CTX_clear_extra_chain_certs(ctx);
++    SSL_CTX_clear_chain_certs(ctx);
+
+     for (int i = 0; i < sk_X509_num(extra_certs); i++) {
+       X509* ca = sk_X509_value(extra_certs, i);
diff --git a/patches/tls-reload-certs-v6.x.patch b/patches/tls-reload-certs-v6.x.patch
new file mode 100644
index 0000000..3976a1a
--- /dev/null
+++ b/patches/tls-reload-certs-v6.x.patch
@@ -0,0 +1,14 @@
+TOOLS-1950 sdcnode 6.x should restore the ability to reload TLS certs at runtime
+diff --git a/src/node_crypto.cc b/src/node_crypto.cc
+index 13af291dfc..b90f014bd6 100644
+--- a/src/node_crypto.cc
++++ b/src/node_crypto.cc
+@@ -493,7 +493,7 @@ int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
+     // the CA certificates.
+     int r;
+
+-    SSL_CTX_clear_extra_chain_certs(ctx);
++    SSL_CTX_clear_chain_certs(ctx);
+
+     for (int i = 0; i < sk_X509_num(extra_certs); i++) {
+       X509* ca = sk_X509_value(extra_certs, i);
diff --git a/patches/tls-reload-certs-v8.x.patch b/patches/tls-reload-certs-v8.x.patch
new file mode 100644
index 0000000..46cbe8d
--- /dev/null
+++ b/patches/tls-reload-certs-v8.x.patch
@@ -0,0 +1,14 @@
+TOOLS-1950 sdcnode 8.x should restore the ability to reload TLS certs at runtime
+diff --git a/src/node_crypto.cc b/src/node_crypto.cc
+index 13af291dfc..b90f014bd6 100644
+--- a/src/node_crypto.cc
++++ b/src/node_crypto.cc
+@@ -688,7 +688,7 @@ int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
+     // the CA certificates.
+     int r;
+
+-    SSL_CTX_clear_extra_chain_certs(ctx);
++    SSL_CTX_clear_chain_certs(ctx);
+
+     for (int i = 0; i < sk_X509_num(extra_certs); i++) {
+       X509* ca = sk_X509_value(extra_certs, i);
